projects/igniteui-angular/src/lib/icon/icon.service.spec.ts

```diff
@@ -152,6 +155,45 @@ describe("Icon Service", () => {
         expect(iconService.isSvgIconCached(iconName, familyName)).toBeTruthy();
     });
 
+    it("should be able to extend family of type font/liga with svg icons", () => {
+        const fixture = TestBed.createComponent(IconTestComponent);
+        const iconName = "test";
+        const familyName = "material";
+
+        iconService.addSvgIconFromText(iconName, svgText, familyName);
+        fixture.detectChanges();
+
+        const extendedIcon = fixture.debugElement.query(By.css("igx-icon[extended] svg")).nativeElement;
+        const builtinIcon = fixture.debugElement.query(By.css("igx-icon[builtin]")).nativeElement;
+
+        expect(extendedIcon).toBeTruthy();
+        expect(builtinIcon).toBeTruthy();
+        expect(svgText).toContain(extendedIcon.innerHTML);
+        expect(builtinIcon.textContent).toContain("home");
+        expect(iconService.defaultFamily.name).toBe(familyName);
+        expect(iconService.defaultFamily.meta.type).toBe("liga");
+    });
+
+    it("should be able to reference icons in extended font/liga families", () => {
+        const fixture = TestBed.createComponent(IconRefComponent);
```

Comment:
Just as a note you could also directly `TestBed.createComponent(IgxIconComponent)` in those simple cases where no extra setup is needed.